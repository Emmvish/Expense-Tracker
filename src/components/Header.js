import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth.js'

const Header = ({startLogout}) => (
  <header className="header">
  	<div className="content-container">
  		<div className="header__content">
    		<Link to="/dashboard" className="header__title">
    			<h1>Expensify</h1>
    		</Link>
    	</div>
    </div>
    <button onClick={startLogout} className="button button--link">Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch)=>({
	startLogout: ()=>dispatch(startLogout)
})

export default connect(undefined,mapDispatchToProps)(Header);

//     <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>