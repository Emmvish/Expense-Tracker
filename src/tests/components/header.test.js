import React from 'react';
import Header from '../../components/Header'
// import ReactShadowRenderer from 'react-test-renderer/shallow'
import {shallow} from 'enzyme';

test('Render Header Component', ()=>{
    // const renderer = new ReactShadowRenderer();
    // renderer.render(<Header/>)
    // Test will always pass the first time as no snapshot of header component exists.
    // Mismatched snapshot of component can be upgraded by pressing 'u' key.
    // expect(renderer.getRenderOutput()).toMatchSnapshot(); 
    const wrapper = shallow(<Header />)
    // expect(weapper.find('h1').text()).toBe('Expensify')
    expect(wrapper).toMatchSnapshot();
})