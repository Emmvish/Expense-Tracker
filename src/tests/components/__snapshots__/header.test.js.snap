// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Component 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create an Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Need Help?
  </NavLink>
</div>
`;
